File: python/tick.py
from json import dumps
from uptime import uptime

TICK_RESOLUTION = (15, 625, 0)
def GetTickCount():
  return int(uptime() * 1000) % (1 << 32)

def GetTickCount64():
  return int(uptime() * 1000)

def ResolveTick(tick):
  resolution = 0
  for i in range(len(TICK_RESOLUTION)):
    resolution = resolution + TICK_RESOLUTION[i] / (10 ** (i * 3))
  if not resolution:
    resolution = 1

  return int((tick * resolution) / resolution) 

if __name__ == "__main__":
  print(dumps({'tick_count': GetTickCount(), 'tick_count_64': GetTickCount64()}))

File: c/tick.c
#include <stdio.h>
#include <stdint.h>
#include <time.h>

#if defined(_WIN32) || defined(WIN32)
#include <sysinfoapi.h>
#else
int32_t GetTickCount() {
  struct timespec ts;
  clock_gettime(CLOCK_MONOTONIC, &ts);
  return (int32_t)(ts.tv_sec * 1000) + (int32_t)(ts.tv_nsec / 1000000);
}

int64_t GetTickCount64() {
  struct timespec ts;
  clock_gettime(CLOCK_MONOTONIC, &ts);
  return (int64_t)(ts.tv_sec * 1000LL) + (int64_t)(ts.tv_nsec / 1000000LL);
}
#endif

void outputJSON() {
  _printf_p(
    "{\"%1$s\": %3$d, \"%2$s\": %4$ld}\n",
    "tick_count",
    "tick_count_64",
    GetTickCount(),
    GetTickCount64()
  );
}

void writeThisAndNextTick() {
  int32_t tick, nextTick;
  tick = GetTickCount();
  do { nextTick = GetTickCount(); } while (tick == nextTick);
  printf("%d -> %d %d\n", CLOCKS_PER_SEC, tick, nextTick);
}

int main() {
  outputJSON();
}